asn1/InterledgerQuotingProtocol.asn

```diff
@@ -2,40 +2,76 @@ InterledgerQuotingProtocol DEFINITIONS AUTOMATIC TAGS ::=
 BEGIN
 
 IMPORTS
-	Address,
-	Amount
-	FROM Types
-	
-	GenericHeaderPreamble
-	FROM GenericExtensionHeaderFormat
+    UInt8,
+    UInt32,
+    UInt64,
+    UInt128,
+    Int8
+    FROM GenericTypes
+
+    Address,
+    Amount,
+    Timestamp,
+    LiquidityCurve
+    FROM InterledgerTypes
 ;
 
-QuoteRequest ::= SEQUENCE {
-	preamble GenericHeaderPreamble,
-	addresses AddressSet,
-	sourceAmount Amount OPTIONAL,
-	destinationAmount Amount OPTIONAL,
-	-- How long the receiver needs to fulfill the payment (when using TTP/ATP)
-	-- Defaults to 1.5 seconds
-	destinationHoldDuration IlpHoldDuration DEFAULT 1500000
+-- Request to receive liquidity information between the current ledger and the
+-- destination account. This information is sufficient to locally quote any
+-- amount until the curve expires.
+QuoteLiquidityRequest ::= SEQUENCE {
+    requestId UInt128,
+    destinationAccount Address,
+    -- How much time the receiver needs to fulfill the payment (in milliseconds)
+    destinationHoldDuration UInt32
+}
+
+QuoteLiquidityResponse ::= SEQUENCE {
+    requestId UInt128,
+    -- Curve describing the liquidity (relationship between input and output
+  -- amounts) for the quoted route
+    liquidity LiquidityCurve,
+    -- Common prefix of all addresses for which this liquidity curve applies
+    -- SHOULD end with a period "." (but clients MUST NOT enforce this)
+    destinationPrefix Address,
+    -- How long the sender should put money on hold (in milliseconds)
+    sourceHoldDuration UInt32
+    -- TODO: An exact quote expiry doesn't seem necessary...
+    -- expiresAt Timestamp
 }
 
-QuoteResponse ::= SEQUENCE {
-	preamble GenericHeaderPreamble,
-	connectorAccount IlpAccountName,
-	-- Address that this quote response relates to
-	address Address,
-	-- Source or destination amount, depending on whichever one was NOT
-	-- provided in the request.
-	requestedAmount Amount,
-	sourceHoldDuration IlpHoldDuration
+-- Quote with a specific source or destination amount
+QuoteBySourceAmountRequest ::= SEQUENCE {
+    requestId UInt128,
+    destinationAccount Address,
+    sourceAmount Amount,
+    -- How much time the receiver needs to fulfill the payment (in milliseconds)
+    destinationHoldDuration UInt32
```

Comment:
Not sure if we should care about that use case. You will always want to hold it for the minimum amount of time. If you don't know how long the recipient needs, you can probably make a reasonable guess or just ask them.

It just doesn't make much sense why you would choose the destination hold based on the sender's willingness to hold. It's like buying paint based on how much they have at the store instead of estimating how much you need.